### PR TITLE
Add @FindBy support for any Component

### PR DESCRIPTION
**Pull request by Toilal**
Work in progress

Related to #229

Test show that it already works for Single element, but fails with FluentList.

### CODE REVIEW

**Comment by Toilal** [comment]
And a refactoring, again :)

This time, it's PageInitializer that have been rewritten to support any kind of objects, fixing #229 at the same time.

@Page as been replaced with @Inject from JSR-330, as it's now possible to use it with any object class, even those that doesn't extends FluentPage. So if one day we implement #213, API will not change because it can be configured to use @Inject annotation.

Also createPage has been renamed by newInstance, as it can now create any type of object instances.

File: fluentlenium-core/src/main/java/org/fluentlenium/core/annotation/Page.java [hunk 1]
```diff
@@ -6,6 +6,10 @@
 import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
 
+/**
+ * @deprecated in favor of {@link javax.inject.Inject}.
+ */
+@Deprecated
```
Comment (filipcynarski) [inline fluentlenium-core/src/main/java/org/fluentlenium/core/annotation/Page.java]: Is it an option to make a Page annotation a reference to Inject just to not force everyone to change theirs annotations in existing frameworks? It seems like huge change to everyone and I'm thinking how to reduce effort on users end.

Comment (Toilal) [inline fluentlenium-core/src/main/java/org/fluentlenium/core/annotation/Page.java]: Currently it's marked deprecated, but works just as ```@Inject``` annotation. I don't plan to remove it for 1.0 release, but maybe for 2.0 (which will be Java8 only).

**Comment by filipcynarski** [comment]
Great work! What is missing here I think is the set of tests of Inject, I can't see anything new for testing how the injection work on unit side. Would you be able to add something before you merge to default?

**Comment by Toilal** [comment]
I still have to write unit test for the ```core.injector``` package.

**Comment by Toilal** [comment]
I just add some unit tests and fixed a few related issues :). I think it's ready to merge now.

**Pull request merged**